runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -73,6 +74,7 @@ ClientSessionData::ClientSessionData(uint64_t clientUID, uint32_t seqNo) :
    TR::SymbolValidationManager::populateSystemClassesNotWorthRemembering(this);
 
    _wellKnownClassesMonitor = TR::Monitor::create("JIT-JITServerWellKnownClassesMonitor");
+   _startupPhaseMonitor = TR::Monitor::create("JIT-JITServerStartupPhaseMonitor");
```

Comment:
I don't think we need a monitor just to get/set a boolean. Such operations are atomic anyway.

Reply:
Are they actually atomic?
This [post](https://stackoverflow.com/questions/16320838/when-do-i-really-need-to-use-atomicbool-instead-of-bool/16321311) says they are not atomic in C++.

Reply:
I am not convinced about the arguments in that post:
> sizeof(bool) is implementation-defined and may conceivably be > 1, so it's possible that it could be non-atomic in some cases.

In theory it's possible to have a bool larger than a `char`, but in practice nobody does that

> You need atomic<bool> to avoid race-conditions. A race-condition occurs if two threads access the same memory location, and at least one of them is a write operation. If your program contains race-conditions, the behavior is undefined

The threads will see either true or false, there is nothing in between. If you wanted to be entirely correct, then you'd have to check the flag and do everything it depends on it in a single critical section, which is impractical. In fact, if another thread updates the flag, I would like to see that update.

> Without atomic there is no guarantee that you'll ever see the update in the other thread at all, or that you'll see updates to variables in the same order that you make them in a different thread. 

This could happen if the value is cached in a register. The use is very far away from where the flag is written, so it's very unlikely that will happen. Even if it happens, start-up phase is just a heuristic, it should not have correctness implications.

Reply:
I see, makes sense. Removed the monitor.